File: alice/paskills/my_alice/src/main/java/ru/yandex/alice/paskills/my_alice/blocks/recommender/RecommenderItem.java
package ru.yandex.alice.paskills.my_alice.blocks.recommender;

import java.util.Objects;
import java.util.Optional;

import javax.annotation.Nullable;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import ru.yandex.alice.paskills.my_alice.layout.Card;
import ru.yandex.alice.paskills.my_alice.layout.PageLayout;

@Data
class RecommenderItem {

    @Nullable
    private final String id;

    @Nullable
    private final String activation;

    @Nullable
    private final String description;

    @Nullable
    @JsonProperty("logo_avatar_id")
    private final String logoAvatarId;

    @Nullable
    @JsonProperty("logo_prefix")
    private final String logoPrefix;

    @Nullable
    private final String look;

    @Nullable
    private final String name;

    @Nullable
    @JsonProperty("logo_bg_image_quasar_id")
    private final String logoBgImageQuasarId;

    @Nullable
    @JsonProperty("logo_fg_round_image_url")
    private final String logoFgRoundImageUrl;

    @Nullable
    @JsonProperty("logo_amelie_bg_url")
    private final String logoAmelieBgUrl;

    @Nullable
    @JsonProperty("search_app_card_item_text")
    private final String searchAppCardItemText;

    @Nullable
    @JsonProperty("logo_my_alice_url")
    private final String logoMyAliceUrl;

    @Nullable
    private final String intro;

    @Nullable
    @JsonProperty("logo_bg_color")
    private final String logoBgColor;

    private boolean isValid() {
        return activation != null && intro != null && description != null && logoMyAliceUrl != null;
    }

    Optional<PageLayout.Card> toCard() {
        if (!isValid()) {
            return Optional.empty();
        }
        return Optional.of(new Card(
                Card.Kind.SCENARIO,
                getActivation(),
                getIntro(),
                getDescription(),
                getLogoMyAliceUrl(),
                Objects.requireNonNullElse(getLogoBgColor(), "#e8eeff"),
                null
        ));
    }

}

File: alice/paskills/dialogovo/common/src/main/java/service/recommender/RecommenderItem.kt
package ru.yandex.alice.paskill.dialogovo.service.recommender

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import ru.yandex.alice.paskill.dialogovo.domain.SkillInfo.Look

@JsonIgnoreProperties(ignoreUnknown = true)
data class RecommenderItem(
    @JsonProperty("id") val skillId: String,
    val activation: String,
    val description: String,
    @JsonProperty("logo_avatar_id") val logoAvatarId: String?,
    @JsonProperty("logo_prefix") val logoPrefix: String,
    val look: Look,
    val name: String,
    @JsonProperty("logo_bg_color") val logoBgColor: String?,
    @JsonProperty("logo_fg_round_image_url") val logoFgRoundImageUrl: String?,
    @JsonProperty("logo_amelie_fg_url") val logoAmelieFgUrl: String?,
    @JsonProperty("logo_amelie_bg_url") val logoAmelieBgUrl: String?,
    @JsonProperty("logo_amelie_bg_wide_url") val logoAmelieBgWideUrl: String?,
    @JsonProperty("search_app_card_item_text") val searchAppCardItemText: String?
)
